### PR TITLE
Binary support for saltpack commands

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco @oconnor663

### CODE REVIEW

File: go/encoding/basex/stream.go [hunk 1]
```diff
@@ -190,6 +188,10 @@ func (d *decoder) Read(p []byte) (int, error) {
 	d.nbuf -= numBytesToDecode
 	copy(d.buf[0:d.nbuf], d.buf[numBytesToDecode:numBytesToDecode+d.nbuf])
 
+	if ret == 0 && d.err == nil && len(p) != 0 {
+		return 0, io.EOF
+	}
```
Comment (oconnor663) [inline go/encoding/basex/stream.go]: Did this fix a bug you ran into?

Comment (maxtaco) [inline go/encoding/basex/stream.go]: Patrick mention something about today on the slack channel

File: go/libkb/stream_classifier.go [hunk 2]
```diff
@@ -127,10 +129,12 @@ type saltpackHeaderPrefix struct {
 
 func isSaltpackBinary(b []byte, sc *StreamClassification) bool {
 	if len(b) < 2 {
+		fmt.Printf("a\n")
```
Comment (oconnor663) [inline go/libkb/stream_classifier.go]: Is this test code that made it in?

Comment (maxtaco) [inline go/libkb/stream_classifier.go]: Yeah looks like it

Comment (patrickxb) [inline go/libkb/stream_classifier.go]: Sorry, will remove the printfs.

File: go/encoding/basex/stream.go (hunk 1 above)
Comment (patrickxb) [inline go/encoding/basex/stream.go]: Yes, see slack and TestBinaryInput in armor_test.go

Comment (patrickxb) [inline go/encoding/basex/stream.go]: There was an infinite loop without this.

**Comment by oconnor663** [comment]
Looks good to me apart from the printfs.

**Comment by patrickxb** [comment]
Thanks, I just removed them.

File: go/saltpack/armor_test.go [hunk 3]
```diff
@@ -143,6 +147,37 @@ func TestSlowReader(t *testing.T) {
...
+		os.Stderr.Write(buf)
+		t.Fatal("timed out waiting for Armor62Open to finish")
+	}
+
+	// PC this seems like a bug to me, but leaving it alone for now:
+	/*
+		if err == nil {
+			t.Errorf("Armor62Open worked on binary data: m == %x, hdr == %q, ftr == %q", m, hdr, ftr)
+		}
+	*/
```
Comment (patrickxb) [inline go/saltpack/armor_test.go]: I'm not crazy about this:  Armor62Open with invalid data returns nil, "", "", nil.

**Pull request merged**